File: solutions/_49_group_anagrams/solution.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function (strs) {
  const res = new Map();
  for (const str of strs) {
    const sortedString = str.split("").sort().join("");
    if (!res.has(sortedString)) {
      res.set(sortedString, []);
    }
    res.get(sortedString).push(str);
  }
  return Array.from(res.values());
};

module.exports = { groupAnagrams };

/*
https://leetcode.com/problems/group-anagrams/
*/

File: solutions/_49_group_anagrams/solution.py
from typing import List
from collections import defaultdict


class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        res = defaultdict(list)
        for s in strs:
            sortedString = "".join(sorted(s))
            res[sortedString].append(s)
        return list(res.values())


# https://leetcode.com/problems/group-anagrams/
